internal/proxy/proxy.go

```diff
@@ -133,32 +143,76 @@ func NewClient(ctx context.Context, d cloudsql.Dialer, cmd *cobra.Command, conf
 		}(inst.Name)
 	}
 	pc := newPortConfig(conf.Port)
+
 	for _, inst := range conf.Instances {
-		m := &socketMount{inst: inst.Name}
-		a := conf.Addr
-		if inst.Addr != "" {
-			a = inst.Addr
-		}
 		version, err := d.EngineVersion(ctx, inst.Name)
 		if err != nil {
 			return nil, err
 		}
-		var np int
-		switch {
-		case inst.Port != 0:
-			np = inst.Port
-		case conf.Port != 0:
-			np = pc.nextPort()
-		default:
-			np = pc.nextDBPort(version)
+
+		var (
+			// network is one of "tcp" or "unix"
+			network string
+			// address is either a TCP host port, or a Unix socket
+			address string
+		)
+		if (conf.UnixSocket != "" || inst.UnixSocket != "") &&
+			(inst.Addr == "" && inst.Port == 0) {
```

Comment:
Do we check anywhere if the user has specified conflicting query parameters?

Reply:
Yes, see above in the diff where we check that an instance query parameter doesn't include both address and unix, or port and unix.